### PR TITLE
Simplify DAG.set_dag_runs_state method

### PR DESCRIPTION
**Pull request by turbaszek**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/models/dag.py [hunk 1]
```diff
@@ -912,12 +912,7 @@ def set_dag_runs_state(
             query = query.filter(DagRun.execution_date >= start_date)
         if end_date:
             query = query.filter(DagRun.execution_date <= end_date)
-        drs = query.all()
-
-        dirty_ids = []
-        for dr in drs:
-            dr.state = state
-            dirty_ids.append(dr.dag_id)
```
Comment (potiuk) [inline airflow/models/dag.py:932]: I wonder why we needed that dirty_ids list. Does not see to be used :).

**Review by potiuk — APPROVED** [review]
Looks good, I just wonder what was the purpose of dirty_ids ?

File: airflow/models/dag.py [hunk 2]
```diff
@@ -912,12 +912,7 @@ def set_dag_runs_state(
...
+        query.update({DagRun.state: state})
```
Comment (kaxil) [inline airflow/models/dag.py:927]: Will this work if multiple DagRuns are returned?

Comment (kaxil) [reply airflow/models/dag.py:927]: I guess yes based on https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.update

**Review by kaxil — APPROVED** [review]
Can you just double-check & confirm that tests for this function are already covered?

File: airflow/models/dag.py (hunk 2 above)
Comment (turbaszek) [reply airflow/models/dag.py:927]: Yes, we are using this in many different places

**Comment by turbaszek** [comment]
> Can you just double-check & confirm that tests for this function are already covered?

@kaxil I've added a test for this method

File: airflow/models/dag.py (hunk 1 above)
Comment (turbaszek) [reply airflow/models/dag.py:932]: That was my main reason to do this change...

Comment (kaxil) [reply airflow/models/dag.py:932]: Yeah that is strange

File: airflow/models/dag.py (hunk 2 above)
Comment (kaxil) [reply airflow/models/dag.py:927]: > > Can you just double-check & confirm that tests for this function are already covered?
> 
> @kaxil I've added a test for this method

Awesome, feel free to merge once the test passes

**Pull request merged**